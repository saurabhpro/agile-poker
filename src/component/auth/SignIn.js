import React from 'react';

import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';

import styles from './styles';

import database, { currentTimeStamp } from '../Firebase';
import { addUserNameToSessionStore } from '../utils/sessionStore';
import { FormControl, Select, MenuItem } from '@material-ui/core';

export default function SignIn({ team }) {
  const [userName, setUserName] = React.useState(undefined);
  const [role, setRole] = React.useState('Team Member');
  const [teamId, setTeamId] = React.useState();

  const addTeam = (event) => {
    setTeamId(event.target.value?.toLowerCase());
  };

  const addName = (event) => {
    setUserName(event.target.value);
  };

  const addRole = (event) => {
    setRole(event.target.value);
  };

  /**
   * async method to fetch the user from firebase
   * and if found set it to be online and active
   */
  async function getAndSetUser() {
    const userRef = database
      .collection('users')
      .doc(userName.toUpperCase());
    const doc = await userRef.get();

    if (!doc.exists) {
      setUser();
    } else {
      setUser(doc.data());
    }
  }

  /**
   * aysnc method to update the firebae document with the actived and online user
   * @param {*} doc the firebase document
   */
  const setUser = async (doc) => {
    const t = team ?? teamId;

    const data = doc
      ? updateUserToOnline(doc)
      : createNewUser(userName.toUpperCase(), role, t);

    // Add a new document in collection "cities" with ID 'LA'
    const res = await database
      .collection('users')
      .doc(userName.toUpperCase())
      .set(data);

    console.log('Set logged in user: ', res);
    // allow refreshing the page - and since our update was sucessful - we wil see home page
    window.location.href = `${process.env.PUBLIC_URL}/${t}`;

    console.log(data);
  };

  /**
   * handles the form submission from the sign in screen
   * @param {*} event
   */
  const onSubmit = (event) => {
    event.preventDefault();

    if (!!userName) {
      getAndSetUser();

      addUserNameToSessionStore(userName);
    }
  };

  /**
   * material ui classes for sign in screen
   */
  const classes = styles();

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h5">
          Agile Poker Table
        </Typography>
        <form className={classes.form} onSubmit={onSubmit} noValidate>
          {!team && (
            <TextField
              variant="outlined"
              margin="normal"
              required
              onChange={addTeam}
              fullWidth
              id="team"
              label="Team"
              name="team"
              autoFocus
            />
          )}
          <TextField
            variant="outlined"
            margin="normal"
            required
            onChange={addName}
            fullWidth
            id="email"
            label="User Name"
            name="userName"
            autoComplete="email"
            autoFocus
          />
          {!team && (
            <FormControl className={classes.formControl} fullWidth>
              <Select
                value={role}
                onChange={addRole}
                defaultValue="Team Member"
                className={classes.selectEmpty}
                inputProps={{ 'aria-label': 'Without label' }}
                variant="outlined"
              >
                <MenuItem value={'Team Member'}>Team Member</MenuItem>
                <MenuItem value={'Product Owner'}>
                  Product Owner
                </MenuItem>
                <MenuItem value={'Scrum Master'}>
                  Scrum Master
                </MenuItem>
              </Select>
            </FormControl>
          )}

          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            Enter
          </Button>
        </form>
      </div>
    </Container>
  );
}

/**
 * Updates the doc to be online and active
 * @param {*} doc
 */
export const updateUserToOnline = (doc) => {
  return {
    userName: doc.userName,
    isOnline: true,
    isActive: true,
    memberSince: doc.memberSince,
    team: doc.team,
    role: doc.role,
  };
};

export const createNewUser = (userName, role, team) => {
  return {
    userName: userName,
    isOnline: true,
    isActive: true,
    memberSince: currentTimeStamp,
    team: team,
    role: role,
  };
};
