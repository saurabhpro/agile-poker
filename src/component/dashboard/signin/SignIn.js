import React from 'react';

import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Box from '@material-ui/core/Box';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';

import styles from './styles';
import Copyright from './Copyright';

export default function SignIn() {
  const [userName, setUserName] = React.useState(undefined);

  const onChange = (event) => {
    setUserName(event.target.value);
  };

  const onSubmit = () => {
    if (userName) {
      sessionStorage.setItem('agilePokerUserName', userName);
    }
  };

  const classes = styles();

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h5">
          Agile Poker Table
        </Typography>
        <form className={classes.form} onSubmit={onSubmit} noValidate>
          <TextField
            variant="outlined"
            margin="normal"
            required
            onChange={onChange}
            fullWidth
            id="email"
            label="Email Address"
            name="userName"
            autoComplete="email"
            autoFocus
          />

          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            Enter
          </Button>
        </form>
      </div>

      <Box mt={8}>
        <Copyright />
      </Box>
    </Container>
  );
}
