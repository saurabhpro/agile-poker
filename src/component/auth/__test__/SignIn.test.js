import React from 'react';
import { mount } from 'enzyme';

import SignIn, { updateUserToOnline } from '../SignIn';


test('should render properly', () => {
  const component = mount(<SignIn />);

  expect(component).toMatchSnapshot();

  component.unmount();
});

describe('sign in a user', () => {
  const user = {
    userName: 'Saurabh',
    isOnline: false,
    isActive: false,
    memberSince: new Date(2020, 20, 8),
    team: 'r2unit',
    role: 'Scrum Master',
  };

  it('it should make user online ', () => {
    //given
    const expected = { ...user, isOnline: true, isActive: true };

    //when
    const actual = updateUserToOnline(user);

    //then
    expect(actual).toStrictEqual(expected);
  });

  it('it should make user active ', () => {
    // given
    expect(user).toStrictEqual({
      ...user,
      isOnline: false,
      isActive: false,
    });

    const expected = { ...user, isOnline: true, isActive: true };

    // when
    const actual = updateUserToOnline(user);

    //then
    expect(actual).toStrictEqual(expected);
  });
});
