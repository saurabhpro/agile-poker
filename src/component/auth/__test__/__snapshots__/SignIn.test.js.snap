// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<SignIn>
  <WithStyles(ForwardRef(Container))
    component="main"
    maxWidth="xs"
  >
    <ForwardRef(Container)
      classes={
        Object {
          "disableGutters": "MuiContainer-disableGutters",
          "fixed": "MuiContainer-fixed",
          "maxWidthLg": "MuiContainer-maxWidthLg",
          "maxWidthMd": "MuiContainer-maxWidthMd",
          "maxWidthSm": "MuiContainer-maxWidthSm",
          "maxWidthXl": "MuiContainer-maxWidthXl",
          "maxWidthXs": "MuiContainer-maxWidthXs",
          "root": "MuiContainer-root",
        }
      }
      component="main"
      maxWidth="xs"
    >
      <main
        className="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <WithStyles(CssBaseline)>
          <CssBaseline
            classes={Object {}}
          />
        </WithStyles(CssBaseline)>
        <div
          className="makeStyles-paper-1"
        >
          <WithStyles(ForwardRef(Avatar))
            className="makeStyles-avatar-2"
          >
            <ForwardRef(Avatar)
              className="makeStyles-avatar-2"
              classes={
                Object {
                  "circle": "MuiAvatar-circle",
                  "colorDefault": "MuiAvatar-colorDefault",
                  "fallback": "MuiAvatar-fallback",
                  "img": "MuiAvatar-img",
                  "root": "MuiAvatar-root",
                  "rounded": "MuiAvatar-rounded",
                  "square": "MuiAvatar-square",
                }
              }
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
              >
                <ForwardRef>
                  <WithStyles(ForwardRef(SvgIcon))>
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </div>
            </ForwardRef(Avatar)>
          </WithStyles(ForwardRef(Avatar))>
          <WithStyles(ForwardRef(Typography))
            component="h1"
            variant="h5"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              component="h1"
              variant="h5"
            >
              <h1
                className="MuiTypography-root MuiTypography-h5"
              >
                Agile Poker Table
              </h1>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
          <form
            className="makeStyles-form-3"
            noValidate={true}
            onSubmit={[Function]}
          >
            <WithStyles(ForwardRef(TextField))
              autoComplete="email"
              autoFocus={true}
              fullWidth={true}
              id="email"
              label="Email Address"
              margin="normal"
              name="userName"
              onChange={[Function]}
              required={true}
              variant="outlined"
            >
              <ForwardRef(TextField)
                autoComplete="email"
                autoFocus={true}
                classes={
                  Object {
                    "root": "MuiTextField-root",
                  }
                }
                fullWidth={true}
                id="email"
                label="Email Address"
                margin="normal"
                name="userName"
                onChange={[Function]}
                required={true}
                variant="outlined"
              >
                <WithStyles(ForwardRef(FormControl))
                  className="MuiTextField-root"
                  color="primary"
                  disabled={false}
                  error={false}
                  fullWidth={true}
                  margin="normal"
                  required={true}
                  variant="outlined"
                >
                  <ForwardRef(FormControl)
                    className="MuiTextField-root"
                    classes={
                      Object {
                        "fullWidth": "MuiFormControl-fullWidth",
                        "marginDense": "MuiFormControl-marginDense",
                        "marginNormal": "MuiFormControl-marginNormal",
                        "root": "MuiFormControl-root",
                      }
                    }
                    color="primary"
                    disabled={false}
                    error={false}
                    fullWidth={true}
                    margin="normal"
                    required={true}
                    variant="outlined"
                  >
                    <div
                      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                    >
                      <WithStyles(ForwardRef(InputLabel))
                        htmlFor="email"
                        id="email-label"
                      >
                        <ForwardRef(InputLabel)
                          classes={
                            Object {
                              "animated": "MuiInputLabel-animated",
                              "asterisk": "MuiInputLabel-asterisk",
                              "disabled": "Mui-disabled",
                              "error": "Mui-error",
                              "filled": "MuiInputLabel-filled",
                              "focused": "Mui-focused",
                              "formControl": "MuiInputLabel-formControl",
                              "marginDense": "MuiInputLabel-marginDense",
                              "outlined": "MuiInputLabel-outlined",
                              "required": "Mui-required",
                              "root": "MuiInputLabel-root",
                              "shrink": "MuiInputLabel-shrink",
                            }
                          }
                          htmlFor="email"
                          id="email-label"
                        >
                          <WithStyles(ForwardRef(FormLabel))
                            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                            classes={
                              Object {
                                "asterisk": "MuiInputLabel-asterisk",
                                "disabled": "Mui-disabled",
                                "error": "Mui-error",
                                "focused": "Mui-focused",
                                "required": "Mui-required",
                              }
                            }
                            data-shrink={false}
                            htmlFor="email"
                            id="email-label"
                          >
                            <ForwardRef(FormLabel)
                              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                              classes={
                                Object {
                                  "asterisk": "MuiFormLabel-asterisk MuiInputLabel-asterisk",
                                  "colorSecondary": "MuiFormLabel-colorSecondary",
                                  "disabled": "Mui-disabled Mui-disabled",
                                  "error": "Mui-error Mui-error",
                                  "filled": "MuiFormLabel-filled",
                                  "focused": "Mui-focused Mui-focused",
                                  "required": "Mui-required Mui-required",
                                  "root": "MuiFormLabel-root",
                                }
                              }
                              data-shrink={false}
                              htmlFor="email"
                              id="email-label"
                            >
                              <label
                                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                                data-shrink={false}
                                htmlFor="email"
                                id="email-label"
                              >
                                Email Address
                                <span
                                  aria-hidden={true}
                                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                                >
                                   
                                  *
                                </span>
                              </label>
                            </ForwardRef(FormLabel)>
                          </WithStyles(ForwardRef(FormLabel))>
                        </ForwardRef(InputLabel)>
                      </WithStyles(ForwardRef(InputLabel))>
                      <WithStyles(ForwardRef(OutlinedInput))
                        autoComplete="email"
                        autoFocus={true}
                        fullWidth={true}
                        id="email"
                        label={
                          <React.Fragment>
                            Email Address
                             *
                          </React.Fragment>
                        }
                        multiline={false}
                        name="userName"
                        onChange={[Function]}
                      >
                        <ForwardRef(OutlinedInput)
                          autoComplete="email"
                          autoFocus={true}
                          classes={
                            Object {
                              "adornedEnd": "MuiOutlinedInput-adornedEnd",
                              "adornedStart": "MuiOutlinedInput-adornedStart",
                              "colorSecondary": "MuiOutlinedInput-colorSecondary",
                              "disabled": "Mui-disabled",
                              "error": "Mui-error",
                              "focused": "Mui-focused",
                              "input": "MuiOutlinedInput-input",
                              "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                              "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                              "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                              "inputMultiline": "MuiOutlinedInput-inputMultiline",
                              "marginDense": "MuiOutlinedInput-marginDense",
                              "multiline": "MuiOutlinedInput-multiline",
                              "notchedOutline": "MuiOutlinedInput-notchedOutline",
                              "root": "MuiOutlinedInput-root",
                            }
                          }
                          fullWidth={true}
                          id="email"
                          label={
                            <React.Fragment>
                              Email Address
                               *
                            </React.Fragment>
                          }
                          multiline={false}
                          name="userName"
                          onChange={[Function]}
                        >
                          <WithStyles(ForwardRef(InputBase))
                            autoComplete="email"
                            autoFocus={true}
                            classes={
                              Object {
                                "adornedEnd": "MuiOutlinedInput-adornedEnd",
                                "adornedStart": "MuiOutlinedInput-adornedStart",
                                "colorSecondary": "MuiOutlinedInput-colorSecondary",
                                "disabled": "Mui-disabled",
                                "error": "Mui-error",
                                "focused": "Mui-focused",
                                "input": "MuiOutlinedInput-input",
                                "inputAdornedEnd": "MuiOutlinedInput-inputAdornedEnd",
                                "inputAdornedStart": "MuiOutlinedInput-inputAdornedStart",
                                "inputMarginDense": "MuiOutlinedInput-inputMarginDense",
                                "inputMultiline": "MuiOutlinedInput-inputMultiline",
                                "marginDense": "MuiOutlinedInput-marginDense",
                                "multiline": "MuiOutlinedInput-multiline",
                                "notchedOutline": null,
                                "root": "MuiOutlinedInput-root",
                              }
                            }
                            fullWidth={true}
                            id="email"
                            inputComponent="input"
                            multiline={false}
                            name="userName"
                            onChange={[Function]}
                            renderSuffix={[Function]}
                            type="text"
                          >
                            <ForwardRef(InputBase)
                              autoComplete="email"
                              autoFocus={true}
                              classes={
                                Object {
                                  "adornedEnd": "MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd",
                                  "adornedStart": "MuiInputBase-adornedStart MuiOutlinedInput-adornedStart",
                                  "colorSecondary": "MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary",
                                  "disabled": "Mui-disabled Mui-disabled",
                                  "error": "Mui-error Mui-error",
                                  "focused": "Mui-focused Mui-focused",
                                  "formControl": "MuiInputBase-formControl",
                                  "fullWidth": "MuiInputBase-fullWidth",
                                  "input": "MuiInputBase-input MuiOutlinedInput-input",
                                  "inputAdornedEnd": "MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd",
                                  "inputAdornedStart": "MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart",
                                  "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                                  "inputMarginDense": "MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense",
                                  "inputMultiline": "MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline",
                                  "inputTypeSearch": "MuiInputBase-inputTypeSearch",
                                  "marginDense": "MuiInputBase-marginDense MuiOutlinedInput-marginDense",
                                  "multiline": "MuiInputBase-multiline MuiOutlinedInput-multiline",
                                  "root": "MuiInputBase-root MuiOutlinedInput-root",
                                }
                              }
                              fullWidth={true}
                              id="email"
                              inputComponent="input"
                              multiline={false}
                              name="userName"
                              onChange={[Function]}
                              renderSuffix={[Function]}
                              type="text"
                            >
                              <div
                                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                                onClick={[Function]}
                              >
                                <input
                                  aria-invalid={false}
                                  autoComplete="email"
                                  autoFocus={true}
                                  className="MuiInputBase-input MuiOutlinedInput-input"
                                  disabled={false}
                                  id="email"
                                  name="userName"
                                  onAnimationStart={[Function]}
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  required={true}
                                  type="text"
                                />
                                <WithStyles(ForwardRef(NotchedOutline))
                                  className="MuiOutlinedInput-notchedOutline"
                                  label={
                                    <React.Fragment>
                                      Email Address
                                       *
                                    </React.Fragment>
                                  }
                                  labelWidth={0}
                                  notched={false}
                                >
                                  <ForwardRef(NotchedOutline)
                                    className="MuiOutlinedInput-notchedOutline"
                                    classes={
                                      Object {
                                        "legend": "PrivateNotchedOutline-legend-6",
                                        "legendLabelled": "PrivateNotchedOutline-legendLabelled-7",
                                        "legendNotched": "PrivateNotchedOutline-legendNotched-8",
                                        "root": "PrivateNotchedOutline-root-5",
                                      }
                                    }
                                    label={
                                      <React.Fragment>
                                        Email Address
                                         *
                                      </React.Fragment>
                                    }
                                    labelWidth={0}
                                    notched={false}
                                  >
                                    <fieldset
                                      aria-hidden={true}
                                      className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                                    >
                                      <legend
                                        className="PrivateNotchedOutline-legendLabelled-7"
                                      >
                                        <span>
                                          Email Address
                                           *
                                        </span>
                                      </legend>
                                    </fieldset>
                                  </ForwardRef(NotchedOutline)>
                                </WithStyles(ForwardRef(NotchedOutline))>
                              </div>
                            </ForwardRef(InputBase)>
                          </WithStyles(ForwardRef(InputBase))>
                        </ForwardRef(OutlinedInput)>
                      </WithStyles(ForwardRef(OutlinedInput))>
                    </div>
                  </ForwardRef(FormControl)>
                </WithStyles(ForwardRef(FormControl))>
              </ForwardRef(TextField)>
            </WithStyles(ForwardRef(TextField))>
            <WithStyles(ForwardRef(Button))
              className="makeStyles-submit-4"
              color="primary"
              fullWidth={true}
              type="submit"
              variant="contained"
            >
              <ForwardRef(Button)
                className="makeStyles-submit-4"
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "containedSizeLarge": "MuiButton-containedSizeLarge",
                    "containedSizeSmall": "MuiButton-containedSizeSmall",
                    "disableElevation": "MuiButton-disableElevation",
                    "disabled": "Mui-disabled",
                    "endIcon": "MuiButton-endIcon",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "iconSizeLarge": "MuiButton-iconSizeLarge",
                    "iconSizeMedium": "MuiButton-iconSizeMedium",
                    "iconSizeSmall": "MuiButton-iconSizeSmall",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                    "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "startIcon": "MuiButton-startIcon",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                    "textSizeLarge": "MuiButton-textSizeLarge",
                    "textSizeSmall": "MuiButton-textSizeSmall",
                  }
                }
                color="primary"
                fullWidth={true}
                type="submit"
                variant="contained"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  type="submit"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    type="submit"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="submit"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Enter
                      </span>
                      <WithStyles(memo)
                        center={false}
                      >
                        <ForwardRef(TouchRipple)
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          >
                            <TransitionGroup
                              childFactory={[Function]}
                              component={null}
                              exit={true}
                            />
                          </span>
                        </ForwardRef(TouchRipple)>
                      </WithStyles(memo)>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
          </form>
        </div>
      </main>
    </ForwardRef(Container)>
  </WithStyles(ForwardRef(Container))>
</SignIn>
`;
